packages/components/Menu/src/MenuTrigger/useMenuTrigger.ts

```diff
@@ -23,7 +24,33 @@ export const useMenuTrigger = (): MenuTriggerState => {
     triggerRef,
   } = context;
 
-  const accessibilityState = open ? expandedState : collapsedState;
+  const {
+    accessibilityActions: childAccessibilityActions,
+    accessibilityState: childAccessibilityState,
+    onAccessibilityAction: childOnAccessibilityAction,
+    onClick: childOnClick,
+    onHoverIn: childOnHoverIn,
+    onHoverOut: childOnHoverOut,
+    componentRef: childComponentRef,
+  } = childProps;
+
+  const accessibilityActions = React.useMemo(() => {
+    if (childAccessibilityActions) {
+      return [...baseAccessibilityActions, ...childAccessibilityActions];
+    }
+
+    return baseAccessibilityActions;
+  }, [childAccessibilityActions]);
+
+  const accessibilityState = React.useMemo(() => {
+    const baseState = open ? expandedState : collapsedState;
+
+    if (childAccessibilityState) {
+      return { ...baseState, ...childAccessibilityState };
+    }
```

Comment:
Dumb question. I noticed that  React Native core doesn't memoize intermediate objects created in a component, but rather memoizes the component itself. Is that something we might want to follow? I wouldn't use them as the example of perfect React code (a lot of it is still legacy..) but thought I'd bring itup.

Reply:
I feel like we should lean more towards general React practices rather than copying what their codebase does... not a strong opinion though.

Reply:
That's fine. This is more me not knowing what the general practice for how much to actually memoize is. Resolving this comment anyway.